api/src/main/java/org/pmiops/workbench/google/DirectoryServiceImpl.java

```diff
@@ -155,17 +157,23 @@ public static void addCustomSchemaValues(User user, String primaryEmail, String
     // Since this value is unlikely to ever change, we use a hard-coded constant rather than an env
     // variable.
     aouCustomFields.put(GSUITE_FIELD_INSTITUTION, INSTITUTION_FIELD_VALUE);
-    // This gives us a structured place to store researchers' contact email addresses, in
-    // case we want to pass it to other systems (e.g. Zendesk or Moodle) via SAML mapped fields.
-    aouCustomFields.put(GSUITE_FIELD_CONTACT_EMAIL, contactEmail);
+
+    if (contactEmail != null) {
+      // This gives us a structured place to store researchers' contact email addresses, in
+      // case we want to pass it to other systems (e.g. Zendesk or Moodle) via SAML mapped fields.
+      aouCustomFields.put(GSUITE_FIELD_CONTACT_EMAIL, contactEmail);
+    }
 
     // In addition to the custom schema value, we store each user's contact email as a secondary
     // email address with type "home". This makes it show up nicely in GSuite admin as the
     // user's "Secondary email".
-    user.setEmails(Lists.newArrayList(
-    new UserEmail().setType("work").setAddress(primaryEmail).setPrimary(true),
-    new UserEmail().setType("home").setAddress(contactEmail)))
-    .setCustomSchemas(Collections.singletonMap(GSUITE_AOU_SCHEMA_NAME, aouCustomFields));
+    ArrayList<UserEmail> emails = Lists.newArrayList(
```

Comment:
nit: prefer the `List` interface.
```suggestion
    List<UserEmail> emails = Lists.newArrayList(
```